torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1892,8 +1896,31 @@ def call_compiled_backward():
                         not backward_state_indices
                     ), "BackwardState requires CompiledAutograd"
                     ctx.maybe_clear_saved_tensors()
+
+                    saved_tensors_use_once = not (
+                        torch.is_grad_enabled() or ctx.is_retain_graph()
```

Comment:
Once https://github.com/pytorch/pytorch/pull/131038, you can change this to `torch._C._autograd._get_current_graph_task_keep_graph()`